flink-connectors/flink-connector-pulsar/src/main/java/org/apache/flink/connector/pulsar/sink/config/PulsarSinkConfigUtils.java

```diff
@@ -70,7 +71,10 @@ public static <T> ProducerBuilder<T> createProducerBuilder(
             PulsarClient client, Schema<T> schema, SinkConfiguration configuration) {
         ProducerBuilder<T> builder = client.newProducer(schema);
 
-        configuration.useOption(PULSAR_PRODUCER_NAME, builder::producerName);
+        configuration.useOption(
```

Comment:
What happens during an application restart? I guess a new producerName is generated. Is this a problem?

Reply:
This wouldn't be a problem. Because we promise consistency by using transactions.